lib/chef-dk/command/generator_commands/cookbook.rb

```diff
@@ -195,9 +203,11 @@ def read_and_validate_params
           @cookbook_name_or_path = arguments[0]
           if !@cookbook_name_or_path
             @params_valid = false
-          elsif /-/ =~ File.basename(@cookbook_name_or_path)
-            err("Hyphens are not allowed in cookbook names. Please specify a cookbook name without hyphens.")
-            @params_valid = false
+          elsif /-/ =~ File.basename(@cookbook_name_or_path) && config[:skip_prompt] == false
+            warn("Hyphens are not allowed in cookbook names.")
```

Comment:
I agree. I don't care for `-` in cookbook names, but if they're not allowed it's news to me. I've seen and used a lot of cookbooks with hyphenated names (with great success, I might add). Encountering a warning that they are not allowed, in opposition to my empirical experience would cause me to question other warnings as possibly inaccurate.